Backend/src/API/REST/routes/team/teamRouter.ts

```diff
@@ -119,8 +124,66 @@ const invite_person_to_a_team = (): Router => {
     })
 
     return router;
-}
+};
 
+/**
+ *  Adds the given projectId to the projects list of the team.
+ */
+const add_project_to_team = (): Router => {
+    let router = express.Router();
+
+    router.put("/teams/:id/add_project", check_auth(), async (req: ExtRequest, res: any) => {
+        try {
+            const {projectId}: {projectId: ObjectId}  = req.body;
+            const teamId: string = req.params.id;
+
+            if(!(projectId && teamId))
+                return res.status(400).send("Missing parameters.");
+
+            const project = await ProjectService.getProjectById(projectId);
+            if (!project)
+                return res.status(400).send("Project to be added does not exist.");
+
+            const team = await TeamService.getTeamById(teamId);
+            if (!team)
+                return res.status(400).send("Team does not exist.");
+
+            /* the user should be either an admin or a member of the team */
+            const isPartOf: boolean = await TeamService.isAdminOrMember(req.user_id!, teamId);
+            if (!isPartOf)
+                return res.status(401).send("Unauthenticated User! The user is not part of the team.");
+
+            const projectOwnerId = await ProjectService.getOwner(projectId); // cannot be null since the project must exist at this point
+            if (projectOwnerId!.toString() !== req.user_id!.toString())
+                return res.status(401).send("User is not the project owner!");
+
+            try {
+                const team_updated = await TeamService.addProject(teamId, projectId);
```

Comment:
TeamService.addProject only adds the projectId to team.projects
but does **not** update the project.teamId.
Is the team.projects necessary? It causes double references.

Reply:
I have added another call in the corresponding route. See `ProjectService.setTeamOfProject`, now the project.teamId is set in this separate call. 

Do you suggest removing the double reference? Can't it be useful in the future e.g. listing the projects of a team?

Reply:
For listing the projects of a team, projects.findMany({teamId}) would also be possible.
Personally, I would remove it, but maybe someone else, maybe @Pellumbengineer could also give his statement.